x-pack/plugins/dataset_quality/public/components/dataset_quality/columns.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiBasicTableColumn, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { PackageIcon } from '@kbn/fleet-plugin/public';
+import { ES_FIELD_TYPES, KBN_FIELD_TYPES } from '@kbn/field-types';
+import { FieldFormatsStart } from '@kbn/field-formats-plugin/public';
+import { DataStreamStat } from '../../../common/data_streams_stats/data_stream_stat';
+
+const nameColumnName = i18n.translate('xpack.datasetQuality.nameColumnName', {
+  defaultMessage: 'Dataset Name',
+});
+
+const sizeColumnName = i18n.translate('xpack.datasetQuality.sizeColumnName', {
+  defaultMessage: 'Size',
+});
+
+const lastActivityColumnName = i18n.translate('xpack.datasetQuality.lastActivityColumnName', {
+  defaultMessage: 'Last Activity',
+});
+
+export const getDatasetQualitTableColumns = ({
+  fieldFormats,
+}: {
+  fieldFormats: FieldFormatsStart;
+}): Array<EuiBasicTableColumn<DataStreamStat>> => {
+  return [
+    {
+      name: nameColumnName,
+      field: 'title',
+      sortable: true,
+      render: (title: string, dataStreamStat: DataStreamStat) => {
+        const { integration } = dataStreamStat;
+
+        return (
+          <EuiFlexGroup alignItems="center" gutterSize="s">
+            {integration ? (
+              <EuiFlexItem grow={false}>
+                <PackageIcon
+                  packageName={integration.name}
+                  version={integration.version!}
+                  icons={integration.icons}
+                  size="m"
+                  tryApi
+                />
+              </EuiFlexItem>
+            ) : (
+              '-'
```

Comment:
We can do that in a follow up PR, for uncategorized datasets we should use this icon
<img width="368" alt="image" src="https://github.com/elastic/kibana/assets/1313018/80706d31-af7e-475d-981e-2716e417b421">